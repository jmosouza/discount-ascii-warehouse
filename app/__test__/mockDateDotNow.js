Date.now = () => new Date('Mon Oct 23 2017 12:00:00 GMT-0000 (-00)').getTime();
