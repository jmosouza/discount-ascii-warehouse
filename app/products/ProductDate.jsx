import React from 'react';
import PropTypes from 'prop-types';

const ProductDate = ({ date }) => (
  <p>{date}</p>
);

ProductDate.propTypes = {
  date: PropTypes.string.isRequired,
};

export default ProductDate;
