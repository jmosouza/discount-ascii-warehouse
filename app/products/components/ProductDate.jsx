import React from 'react';
import PropTypes from 'prop-types';
import relativeDateInterval from '../../utils/relativeDateInterval';

const ProductDate = ({ date }) => (
  <p>{relativeDateInterval({ dateStr: date })}</p>
);

ProductDate.propTypes = {
  date: PropTypes.string.isRequired,
};

export default ProductDate;
