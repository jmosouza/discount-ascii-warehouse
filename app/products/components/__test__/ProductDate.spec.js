/* global test, expect */
import renderer from 'react-test-renderer';
import React from 'react';
import ProductDate from '../ProductDate';
import mockDateDotNow from '../../../__test__/mockDateDotNow';

mockDateDotNow();

[
  'Mon Oct 23 2017 13:00:00 GMT-0000 (-00)',
  'Mon Oct 23 2017 11:59:50 GMT-0000 (-00)',
  'Mon Oct 23 2017 11:59:30 GMT-0000 (-00)',
  'Mon Oct 23 2017 11:59:00 GMT-0000 (-00)',
  'Mon Oct 23 2017 11:30:00 GMT-0000 (-00)',
  'Mon Oct 23 2017 11:00:00 GMT-0000 (-00)',
  'Mon Oct 23 2017 06:00:00 GMT-0000 (-00)',
  'Sun Oct 22 2017 12:00:00 GMT-0000 (-00)',
  'Sat Oct 21 2017 12:00:00 GMT-0000 (-00)',
  'Fri Oct 20 2017 12:00:00 GMT-0000 (-00)',
  'Fri Oct 16 2017 12:00:00 GMT-0000 (-00)',
].forEach((dateStr) => {
  test('ProductDate', () => {
    expect(renderer.create(
      <ProductDate
        date={dateStr}
      />,
    )).toMatchSnapshot();
  });
});
