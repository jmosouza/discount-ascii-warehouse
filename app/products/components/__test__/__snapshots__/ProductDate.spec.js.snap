// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDate 1`] = `
<p>
  comming soon
</p>
`;

exports[`ProductDate 2`] = `
<p>
  just then
</p>
`;

exports[`ProductDate 3`] = `
<p>
  30 seconds ago
</p>
`;

exports[`ProductDate 4`] = `
<p>
  a minute ago
</p>
`;

exports[`ProductDate 5`] = `
<p>
  30 minutes ago
</p>
`;

exports[`ProductDate 6`] = `
<p>
  1 hour ago
</p>
`;

exports[`ProductDate 7`] = `
<p>
  6 hours ago
</p>
`;

exports[`ProductDate 8`] = `
<p>
  yesterday
</p>
`;

exports[`ProductDate 9`] = `
<p>
  2 days ago
</p>
`;

exports[`ProductDate 10`] = `
<p>
  3 days ago
</p>
`;

exports[`ProductDate 11`] = `
<p>
  10/16/2017
</p>
`;
